nav2_graceful_controller/src/graceful_controller.cpp

```diff
@@ -177,7 +180,7 @@ geometry_msgs::msg::TwistStamped GracefulController::computeVelocityCommands(
       next_pose.pose.orientation = nav2_util::geometry_utils::orientationAroundZAxis(yaw);
       geometry_msgs::msg::PoseStamped costmap_pose;
       tf2::doTransform(next_pose, costmap_pose, costmap_transform);
-      if (inCollision(
+      if (use_collision_detection && inCollision(
```

Comment:
```suggestion
      if (params_->use_collision_detection && inCollision(
```

And remove L174 where `use_collision_detection` is set, that's not needed